core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -195,6 +197,11 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
           makeOffers()
         }
 
+      // Only be used for testing.
+      case ReviveOffers =>
```

Comment:
If you need to expose this for testing, I'd prefer to make makeOffers() private[scheduler] (and then the test can directly call backend.makeOffers())